src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Web/Debugger/WebLaunchSettingsProvider.cs

```diff
@@ -67,12 +67,12 @@ public async Task<WebLaunchSettings> GetLaunchSettingsAsync()
 
             if (_flavorServerSettings == null)
             {
-                return new WebLaunchSettings(serverType, urls, _useWindowsAuth, _useAnonymousAuth, _useClassicPipelineMode, _use64BitIISExpress, _useGlobalApplicationHostFile, false, null);
+                return new WebLaunchSettings(serverType, urls, _useWindowsAuth, _useAnonymousAuth, _useClassicPipelineMode, _use64BitIISExpress, _useGlobalApplicationHostFile);
             }
 
             if (_flavorServerSettings.UseCustomServer)
             {
-                if (_flavorServerSettings.CustomServerUrl != null)
+                if (_flavorServerSettings.CustomServerUrl != null && _flavorServerSettings.CustomServerUrl.Length != 0)
```

Comment:
`!string.IsNullOrEmpty` (and below)?

Reply:
Here was my comment to Saint:

> Do you know how to avoid the "CS8604: Possible Null reference argument" I get on the urls.Add line? The analyzer is not smart enough to recognize that string.IsNullOrEmpty is a null check. I realize I can pragma disable it but that is pretty ugly.
> if (!string.IsNullOrEmpty(_flavorServerSettings.IISUrl))
> {
> urls.Add(_flavorServerSettings.IISUrl);
> 
> Or I guess I can do the checks myself: ie
> if (_flavorServerSettings.IISUrl) != null && _flavorServerSettings.IISUrl.Length != 0)

Reply:
Sorry I missed that discussion.

You're right. In .NET Framework, `string.IsNullOrEmpty`/`WhiteSpace` are not annotated for nullability.

To work around this, we have our own annotated versions that delegate and which would hopefully be inlined by the JIT. If you use `Strings.IsNullOrEmpty`, the nullability analysis will work as you would expect.

https://github.com/dotnet/project-system/blob/a33291163b1aae554a2657d52aa5b43eb5fc3f09/src/Microsoft.VisualStudio.ProjectSystem.Managed/Utilities/Strings.cs#L9-L10

Reply:
Perfect, thanks